contrib/ibm-mq/main.go

```diff
@@ -0,0 +1,183 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/caarlos0/env"
+	"github.com/ibm-messaging/mq-golang/ibmmq"
+	"knative.dev/pkg/cloudevents"
+)
+
+var sink string
+var wg sync.WaitGroup
+
+type config struct {
+	QueueManager   string `env:"QUEUE_MANAGER" envDefault:"QM1"`
+	ChannelName    string `env:"CHANNEL_NAME" envDefault:"DEV.APP.SVRCONN"`
+	ConnectionName string `env:"CONNECTION_NAME" envDefault:"localhost(1414)"`
+	UserID         string `env:"USER_ID" envDefault:"app"`
+	Password       string `env:"PASSWORD" envDefault:"password"`
+	QueueName      string `env:"QUEUE_NAME" envDefault:"DEV.QUEUE.1"`
+}
+
+//MQMessage wraps message descriptor and message body into one struct
+type MQMessage struct {
+	Message     *ibmmq.MQMD `json:"message_descriptor"`
+	MessageData string      `json:"message_data"`
+}
+
+func init() {
+	flag.StringVar(&sink, "sink", "http://localhost:8080", "where to sink events to")
+}
+
+func main() {
+	flag.Parse()
+
+	cfg := config{}
+	if err := env.Parse(&cfg); err != nil {
+		log.Fatal(err)
+	}
+
+	cloudEventsClient := cloudevents.NewClient(
+		sink,
+		cloudevents.Builder{
+			Source:    "ibm:mq",
+			EventType: "message queue item",
+		},
+	)
+
+	fmt.Printf("%+v\n", cfg)
+
+	// create IBM MQ channel definition
+	channelDefinition := ibmmq.NewMQCD()
+	channelDefinition.ChannelName = cfg.ChannelName
+	channelDefinition.ConnectionName = cfg.ConnectionName
+
+	// init connection security params
+	connSecParams := ibmmq.NewMQCSP()
+	connSecParams.AuthenticationType = ibmmq.MQCSP_AUTH_USER_ID_AND_PWD
+	connSecParams.UserId = cfg.UserID
+	connSecParams.Password = cfg.Password
+
+	// setup MQ connection params
+	connOptions := ibmmq.NewMQCNO()
+	connOptions.Options = ibmmq.MQCNO_CLIENT_BINDING
+	// connOptions.ApplName = "Triggermesh eventing"
+	connOptions.ClientConn = channelDefinition
+	connOptions.SecurityParms = connSecParams
+
+	// And now we can try to connect. Wait a short time before disconnecting.
+	qMgrObject, err := ibmmq.Connx(cfg.QueueManager, connOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Printf("Connection to %s succeeded.\n", cfg.QueueManager)
+	defer disconnect(qMgrObject)
+
+	// Create the Object Descriptor that allows us to give the queue name
+	mqod := ibmmq.NewMQOD()
+
+	// We have to say how we are going to use this queue. In this case, to GET
+	// messages. That is done in the openOptions parameter.
+	openOptions := ibmmq.MQOO_INPUT_SHARED
+
+	// Opening a QUEUE (rather than a Topic or other object type) and give the name
+	mqod.ObjectType = ibmmq.MQOT_Q
+	mqod.ObjectName = cfg.QueueName
+
+	qObject, err := qMgrObject.Open(mqod, openOptions)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("Opened queue", qObject.Name)
```

Comment:
same here

Reply:
thanks @lionelvillard, I've entirely got rid of fmt methods in the app